### ISSUE TITLE
bug(no-any): incorrect suggestion

### ISSUE THREAD

**Issue by andreww2012** [marcalexiei/eslint-plugin-zod#143]
### 🧩 Description

If there are any chained methods after `.any()`, the suggestion to replace `any` with `unknown` always replaces the last method in the chain, despite it not being `any`.

### 🪜 How to reproduce

This code:

```ts
import * as z from 'zod';

export const Schema = z.any().refine((value) => value)
```

gets transformed by the suggestion to:

```ts
import * as z from 'zod';

export const Schema = z.any().unknown((value) => value)
```

### ✅ Expected behavior

The correct fix is

```ts
import * as z from 'zod';

export const Schema = z.unknown().refine((value) => value)
```

### 💻 System info

```shell
npmPackages:
    eslint: 9.39.1 => 9.39.1
    eslint-plugin-zod-x: 1.12.0 => 1.12.0
```
